### ISSUE TITLE
`error[E0433]: failed to resolve: could not find 'mount' in 'backend'`

### ISSUE THREAD

**Issue by polarathene** [bytecodealliance/rustix#1193]
**TL;DR:** I think this was meant to be removed from `lib.rs` with [this Sep 2024 PR](https://github.com/bytecodealliance/rustix/pull/1154), but was missed?

https://github.com/bytecodealliance/rustix/blob/ec96121bf66b065452bf2d784035400df558aabd/src/lib.rs#L296-L300

---

I wanted to try the recently implemented `select` method, and realized it's not available in a release yet, so I switched to git for `rustix`:

```toml
rustix = { git = "https://github.com/bytecodealliance/rustix.git", features = ["procfs", "event"] }
```

Compiling fails however due to various errors about the backend lacking the `mount` module (_All errors are from `src/mount/` files referring to the backend mount module_):

```rust
error[E0433]: failed to resolve: could not find `mount` in `backend`
  --> /root/.cargo/git/checkouts/rustix-8fafe793515fde97/ec96121/src/mount/fsopen.rs:3:21
   |
3  | use crate::backend::mount::types::{
   |                     ^^^^^ could not find `mount` in `backend`
   |
note: found an item that was configured out
  --> /root/.cargo/git/checkouts/rustix-8fafe793515fde97/ec96121/src/backend/linux_raw/mod.rs:49:16
   |
49 | pub(crate) mod mount;
   |                ^^^^^
note: the item is gated behind the `mount` feature
  --> /root/.cargo/git/checkouts/rustix-8fafe793515fde97/ec96121/src/backend/linux_raw/mod.rs:48:7
   |
48 | #[cfg(feature = "mount")]
   |       ^^^^^^^^^^^^^^^^^
```

The `fs` feature is implicitly enabled from `procfs` feature (_so `lib.rs` then enables the `mount` module_):

https://github.com/bytecodealliance/rustix/blob/ec96121bf66b065452bf2d784035400df558aabd/Cargo.toml#L161-L162

## Context

- Earlier PR: https://github.com/bytecodealliance/rustix/pull/763
- Recent Sep 2024 PR (_[removed for backends](https://github.com/bytecodealliance/rustix/pull/1154/files#diff-19da5b08da9ce4980b74a43f8ad55fe785de57fa3c9e0117db639962e59fb27a), but not `lib.rs`_): https://github.com/bytecodealliance/rustix/pull/1154

**NOTE:** [This comment line was removed](https://github.com/bytecodealliance/rustix/commit/f6f19e06ddff893dd9bb511816f370c78ca40d0a#diff-89abf36b9343684b289770007f4b6d667b2940103f05d5a1ccc46dafa3ae1cf5L9), not sure if that context was intended to be dropped along with the deprecation comment above it.

Config attributes for `mod mount`:

https://github.com/bytecodealliance/rustix/blob/ec96121bf66b065452bf2d784035400df558aabd/src/lib.rs#L217-L220

https://github.com/bytecodealliance/rustix/blob/ec96121bf66b065452bf2d784035400df558aabd/src/lib.rs#L296-L300

https://github.com/bytecodealliance/rustix/blob/ec96121bf66b065452bf2d784035400df558aabd/src/lib.rs#L176-L180

https://github.com/bytecodealliance/rustix/blob/ec96121bf66b065452bf2d784035400df558aabd/src/backend/linux_raw/mod.rs#L48-L49

**Comment by sunfishcode** [comment]
Thanks for tracking this down!
